pkg/clustermanager/upgrader.go

```diff
@@ -54,3 +54,39 @@ func EksaChangeDiff(currentSpec, newSpec *cluster.Spec) *types.ChangeDiff {
 	}
 	return nil
 }
+
+func (u *Upgrader) EksdUpgrade(ctx context.Context, cluster *types.Cluster, currentSpec, newSpec *cluster.Spec) (*types.ChangeDiff, error) {
```

Comment:
can we move this to a new entity? like `eksd.Upgrader`
Ideally we should stop adding more things to `clustermanager`, it already has too many responsibilities.